Repository Cleanup

nachfq opened the pull request:
# Repository Structure Optimization

## Project Organization & Configuration

✅ **Centralized Configuration Management** - Streamlined project configuration

- config/abis.ts (added)
- config/constants.ts (added)
- config/deployment-config.ts (added)
- config/networks.ts (added)

✅ **ABI Management Restructure** - Organized external contract ABIs

- src/abis/arbWasmCacheABI.json → abis/external/arbWasmCache.abi.json (moved)
- src/abis/cacheManagerABI.json → abis/external/cacheManager.abi.json (moved)

## Contract Architecture Improvements

### Core Contract Consolidation

✅ **Unified CacheManagerAutomation** - Simplified contract structure

- src/contracts/CacheManagerAutomation.sol (deleted)
- src/contracts/CacheManagerAutomationV2.sol → contracts/core/CacheManagerAutomation.sol (moved)
- src/contracts/BiddingEscrow.sol → contracts/core/BiddingEscrow.sol (moved)

### Interface Optimization

✅ **Streamlined Interface Structure** - Clean interface definitions

- src/contracts/interfaces/ICacheManagerAutomation.sol (deleted)
- src/contracts/interfaces/ICacheManagerAutomationV2.sol → contracts/interfaces/ICacheManagerAutomation.sol (moved)
- src/contracts/interfaces/IExternalContracts.sol → contracts/interfaces/IExternalContracts.sol (moved)

## Development Workflow Enhancements

### Script Organization

✅ **Structured Script Management** - Organized deployment and utility scripts

- scripts/deploy/deploy-cache-manager-automation.ts (added)
- scripts/utils/generate-types.ts (added)
- scripts/utils/verify-abis.ts (added)

### Testing Infrastructure

✅ **Consolidated Test Structure** - Unified testing approach

- test/CacheManagerAutomationV2.test.ts (deleted)
- test/CacheManagerAutomation.test.ts (modified)
- test/scripts/monitor.ts (deleted)
- test/scripts/monitorNoFilter.ts (deleted)
- test/db/.dummy (deleted)

### Utility Scripts Reorganization

✅ **Enhanced Development Tools** - Improved development utilities

- test/scripts/deploy-dummy-wasm.sh → test/utils/deploy-dummy-wasm.sh (moved)
- test/scripts/place-bids.sh → test/utils/place-bids.sh (moved)
- test/utils/cma-cheatsheet.sh (added)

## Configuration & Environment

### Environment Management

✅ **Simplified Environment Configuration** - Cleaner environment setup

- .env.sepolia (deleted)
- .env.example (modified)
- .gitignore (updated)

### Build Configuration

✅ **Hardhat Configuration Enhancement** - Improved build system

- hardhat.config.ts (modified)
- foundry.toml (deleted)
- package.json (updated)

## ARCHITECTURE IMPROVEMENTS

### Configuration-Driven Architecture

✅ **Centralized Configuration System** - All project configurations in dedicated config directory

- Network configurations with proper typing
- Deployment configurations per environment
- External contract ABI management
- Constants and addresses centralization

### Modular Contract Structure

✅ **Simplified Contract Hierarchy** - Clear separation of concerns

- Core contracts in contracts/core/
- Interfaces in contracts/interfaces/
- Clean import structure

### Enhanced Development Experience

✅ **Improved Developer Workflow** - Better tooling and organization

- Structured script organization
- Unified testing approach
- Enhanced type generation
- ABI verification utilities

nachfq committed:
Repo Cleanup

- Updated .env.example to include configurations for both local and Sepolia environments, providing clearer examples for users.
- Removed the obsolete .env.sepolia file to streamline environment management.
- Enhanced the CacheManagerAutomation contract by removing the deprecated AutomationCompatibleInterface and updating internal logic for better performance.
- Updated hardhat.config.ts and foundry.toml to reflect new build paths and configurations.
- Cleaned up test files and removed unused scripts to improve project organization.

nachfq committed:
Repo Cleanup

nachfq committed:
Update .gitignore to include addresses.txt

nachfq committed:
Remove foundry.toml configuration file

nachfq committed:
Refactor Hardhat configuration and enhance project structure

- Consolidated network configurations into a separate `config/networks.ts` file for better organization.
- Updated `hardhat.config.ts` to utilize the new network configuration structure.
- Added `constants.ts` and `deployment-config.ts` for managing contract addresses and deployment settings.
- Introduced `BiddingEscrow` and `CacheManagerAutomation` contracts with comprehensive functionality for user fund management and bidding automation.
- Enhanced `README.md` to reflect the new project structure and provide clearer installation and usage instructions.
- Updated `package.json` with additional scripts for deployment and testing.
- Implemented interfaces for external contracts to improve modularity and maintainability.

nachfq committed:
Update project structure and enhance ABI management

- Added new scripts for generating and verifying ABIs, and for generating TypeScript types from ABIs in package.json.
- Updated .gitignore to exclude generated files.
- Enhanced README.md with detailed ABI management instructions and project structure.
- Refactored CacheManagerAutomation contract to remove upgradeable patterns, simplifying the architecture.
- Deleted obsolete ABI JSON files and updated import paths in test helpers for better organization.

nachfq committed:
Refactor ABI management and update project structure

- Removed obsolete ABI files and updated the package.json scripts to streamline the build process.
- Enhanced README.md to clarify the new structure for external contract management and TypeScript type generation.
- Refactored the ABI and TypeScript generation scripts to focus solely on external contracts, improving organization and maintainability.
- Updated the configuration for ABI verification to reflect the changes in the project structure.

nachfq committed:
Refactor CacheManagerAutomation tests and deployment logic

- Removed the upgradable proxy pattern from the CacheManagerAutomation deployment in favor of a direct deployment approach, simplifying the architecture.
- Updated related test cases to reflect changes in contract deployment and ensure accurate contract management functionality.
- Cleaned up test descriptions and assertions for clarity and correctness.

ifqbuilder commented on test/utils/cma-cheatsheet.sh:
```diff
@@ -0,0 +1,73 @@
+
+ARB_WASM_CACHE_ADDRESS=0x0000000000000000000000000000000000000072
+USER_ADDRESS=0x39DaE6A77A5165598aEB84cAe96Aea0A2215bCa8
+USER_PK=0x6ec659638309dd9b18ded7cab5a3c70b0ff30ae72aa3e6440ea3d79ef375933c
```
Son mias de prueba, no hay problema.

Igual las borro asi queda mas prolijo

ifqbuilder commented on config/constants.ts:
```diff
@@ -0,0 +1,26 @@
+};
+
+export const ARB_WASM_CACHE_ADDRESSES = {
+  arbitrumSepolia: '0x0000000000000000000000000000000000000072', // Replace with actual address
```
La address de los precompiles es la misma en todas las chains!
Ese comment me quedo viejo. actualizo!

nachfq committed:
Update contract addresses and deployment configurations for Arbitrum networks

- Added Arbitrum One addresses to CACHE_MANAGER_ADDRESSES and ARB_WASM_CACHE_ADDRESSES.
- Updated deployment configurations to include Arbitrum One settings.
- Removed commented-out code related to BiddingEscrow deployment in the automation script for clarity.
- Cleaned up environment variable definitions in the cheatsheet script for better organization.
